snuba/querylog/__init__.py

```diff
@@ -90,6 +93,37 @@ def _record_bytes_scanned_metrics(
     metrics.increment("progress_bytes", progress_bytes, tags)
 
 
+def _record_cogs(
+    request: Request,
+    query_metadata: SnubaQueryMetadata,
+    result: Union[QueryResult, QueryException, QueryPlanException],
+) -> None:
+    """
+    Record bytes scanned for Generic Metrics Queries per use case.
+    """
+
+    if (
+        not isinstance(result, QueryResult)
+        or query_metadata.dataset != "generic_metrics"
+    ):
+        return
+
+    profile = result.result.get("profile")
+    if not profile or (bytes_scanned := profile.get("progress_bytes")) is None:
+        return
+
+    if (use_case_id := request.attribution_info.tenant_ids.get("use_case_id")) is None:
```

Comment:
Who is assigning `use_case_id` ?

Reply:
Eventually all `generic_metrics` queries should have a `use_case_id` Tenant ID, there's not really any other way to figure out what use case to attribute to a generic metrics query.

Example labels I've started adding myself [here](https://github.com/getsentry/sentry/pulls?q=is%3Apr+author%3Arahul-kumar-saini+CoGS)